File: caco3_fortran.py
# -*- coding: utf-8 -*-
"""
signal tracking diagenesis model
"""
import os
if os.path.exists('./caco3mod.dll'):import caco3mod

ccflxi      = 12e-6                 # caco3 rain flux in mol cm-2 yr-1
om2cc       = 0.7                   # om/caco3 rain ratio 
dep         = 3.5                   # water depth in km 
dtinput     = 1e2                   # time step in yr 
runname     = 'test'                # file name 
oxonly      = False                 # oxic-only OM degradation model or not 
biotmode    = 'fickian'             # biomixing styles, choose from 'fickian','nobio','turbo2','labs'

caco3mod.caco3_python(ccflxi,om2cc,dep,dtinput,runname,oxonly,biotmode) 

File: caco3_fortran.f90
program caco3_fort
implicit none 
! contains 
real(kind=8)::ccflxi,om2cc,dtinput,dep,ztot
character*255::runname,biotmode,co2chem,runmode
logical::oxonly

call getinput_v2(ccflxi,om2cc,dtinput,runname,dep,oxonly,biotmode)
call caco3(ccflxi,om2cc,dep,dtinput,runname,oxonly,biotmode) 

endprogram 
